src/app/home-page/components/home-page/home-page.component.ts

```diff
@@ -139,14 +23,18 @@ export class HomePageComponent {
     htmlSnippet: '<img style="width:100%" src="../../../../assets/img/adv_area.png" >',
   }];
 
-  wishListFilter = this.products.filter(el => el.addedToWishList);
 
-  refreshWishList(resFromChild) {
-    if (this.wishListFilter.every(el => el.id !== resFromChild.id)) {
-      this.wishListFilter.unshift(resFromChild);
-      return;
-    }
-      const indexOfItem = this.wishListFilter.findIndex(el => el.id === resFromChild.id);
-      this.wishListFilter.splice(indexOfItem, 1);
+  constructor(
+    public productItemService: ProductItemService
+  ) { }
+
+  ngOnInit(): void {
+    this.productItemService.getProducts().subscribe(data => {
+      this.products.push(data);
+    });
+  }
+
+  public wishListHandler(product): void {
```

Comment:
create addToWishList and removeFromWishList

handle error if product no longer exists

Reply:
done